core/serialization_v6.c

```diff
@@ -1049,11 +1050,11 @@ uint32_t chunk_v6_read_shape(Stream *s,
 
                 nameStr = (char *)malloc(nameLen + 1); // +1 for null terminator
                 if (nameStr == NULL) {
-                    totalSizeRead += sizeRead + (uint32_t)sizeof(uint32_t);
-                    continue;
+                    cclog_error("malloc failed");
```

Comment:
Similarly to how it's done for `P3S_CHUNK_ID_SHAPE_POINT` chunk above, we should check if `nameStr` is NULL before calling `map_string_float3_set_key_value` below (line 1069)

Reply:
done